lib/taskana-core/src/main/java/pro/taskana/task/internal/TaskServiceImpl.java

```diff
@@ -617,6 +619,55 @@ public void forceDeleteTask(String taskId)
     }
   }
 
+  @Override
+  public BulkOperationResults<String, TaskanaException> setOwnerOfTasks(
+      String owner, List<String> argTaskIds) {
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug(
+          "entry to setOwnerOfTasks(owner = {}, tasks = {})",
+          owner,
+          LoggerUtils.listToString(argTaskIds));
+    }
+    try {
+      taskanaEngine.openConnection();
```

Comment:
We could move this down to do the preparations before we open the connection. This way we reduce the time the DB connection is used.

Reply:
Done